ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -514,6 +514,28 @@ DiskManager addDisk(DiskId diskId) {
     });
   }
 
+  /**
+   * Add a new store to the storage manager for file copy based replication post filecopy is completed.
+   * @param replica the {@link ReplicaId} of the {@link Store} which would be added.
+   * @return
+   */
+  @Override
+  public boolean addBlobStoreForFileCopy(ReplicaId replica) {
+    if (partitionToDiskManager.containsKey(replica.getPartitionId())) {
```

Comment:
This will get added as part of file store, so it might occur that the this always returns true.

Reply:
We should instead check below map for existence of store in disk manager.
  private final ConcurrentHashMap<PartitionId, BlobStore> stores = new ConcurrentHashMap<>();

Reply:
Yes, diskmanager will be started already in pre-filecopy steps. Instead, I have put the check of partitionToDiskManager not having the partitionId key and fetching the diskManager from this map itself. Checking of store's existence is handled in the addBlobStoreForFileCopy call.